app/lib/kafka.server.ts

```diff
@@ -13,11 +13,13 @@ import { domain, getEnvOrDieInProduction } from './env.server'
 
 const client_id = getEnvOrDieInProduction('KAFKA_CLIENT_ID') ?? ''
 const client_secret = getEnvOrDieInProduction('KAFKA_CLIENT_SECRET')
-const kafka = new Kafka({
-  client_id,
-  client_secret,
-  domain,
-})
+const kafka = process.env.CI
+  ? undefined
+  : new Kafka({
+      client_id,
+      client_secret,
+      domain,
+    })
```

Comment:
Why doesn't this work in CI/CD?

Reply:
I think it is missing env variables for  `KAFKA_CLIENT_ID`, `KAFKA_CLIENT_SECRET`, and possibly origin, since the default for the `domain` value is test if it is not present. Do we want to be producing messages in the ci/cd? If so, I can just revert these changes and add some credentials to make this work

Reply:
The local development environment should work even if the user has not configured any environment variables.

Please remove the `process.env.CI` check. Instead, you should fail gracefully and log a warning if the Kafka credentials are not configured.

Reply:
Come to think of it, a graceful fallback path for the Kafka credentials would be nice as a separate PR.